packages/auth-construct/src/construct.ts

```diff
@@ -900,132 +902,209 @@ export class AmplifyAuth
+      produce: () => {
+        const usernameAttributes: string[] =
+          cfnUserPool.usernameAttributes ?? [];
+        return JSON.stringify(
+          usernameAttributes.map((attr) => attr.toLowerCase())
+        );
+      },
+    });
 
-    if (this.computedUserPoolProps.autoVerify) {
-      const verificationMechanisms = [];
-      if (this.computedUserPoolProps.autoVerify.email) {
-        verificationMechanisms.push('email');
-      }
-      if (this.computedUserPoolProps.autoVerify.phone) {
-        verificationMechanisms.push('phone_number');
-      }
-      if (verificationMechanisms.length > 0) {
-        output.verificationMechanisms = JSON.stringify(verificationMechanisms);
-      }
-    }
+    // extract verificationMechanisms from UserPool's autoVerifiedAttributes
+    output.verificationMechanisms = Lazy.string({
+      produce: () => {
+        const verificationMechanism: string[] = [];
+        (cfnUserPool.autoVerifiedAttributes ?? []).forEach((attr) => {
+          verificationMechanism.push(attr);
+        });
+        return JSON.stringify(verificationMechanism);
+      },
+    });
 
-    if (this.computedUserPoolProps.passwordPolicy) {
-      output.passwordPolicyMinLength =
-        this.computedUserPoolProps.passwordPolicy.minLength?.toString();
+    // extract the passwordPolicy from the UserPool policies
+    output.passwordPolicyMinLength = Lazy.string({
+      produce: () => {
+        if (cfnUserPool.policies) {
+          const policy = (cfnUserPool.policies as CfnUserPool.PoliciesProperty)
+            .passwordPolicy as CfnUserPool.PasswordPolicyProperty;
+          return policy.minimumLength?.toString();
+        }
+        return undefined;
+      },
+    });
 
-      const requirements = [];
-      if (this.computedUserPoolProps.passwordPolicy.requireDigits) {
-        requirements.push('REQUIRES_NUMBERS');
-      }
-      if (this.computedUserPoolProps.passwordPolicy.requireLowercase) {
-        requirements.push('REQUIRES_LOWERCASE');
-      }
-      if (this.computedUserPoolProps.passwordPolicy.requireUppercase) {
-        requirements.push('REQUIRES_UPPERCASE');
-      }
-      if (this.computedUserPoolProps.passwordPolicy.requireSymbols) {
-        requirements.push('REQUIRES_SYMBOLS');
-      }
+    output.passwordPolicyRequirements = Lazy.string({
+      produce: () => {
+        if (cfnUserPool.policies) {
+          const policy = (cfnUserPool.policies as CfnUserPool.PoliciesProperty)
+            .passwordPolicy as CfnUserPool.PasswordPolicyProperty;
+          const requirements = [];
+          policy.requireNumbers && requirements.push('REQUIRES_NUMBERS');
+          policy.requireLowercase && requirements.push('REQUIRES_LOWERCASE');
+          policy.requireUppercase && requirements.push('REQUIRES_UPPERCASE');
+          policy.requireSymbols && requirements.push('REQUIRES_SYMBOLS');
+          return JSON.stringify(requirements);
+        }
+        return undefined;
```

Comment:
```suggestion
        if (!cfnUserPool.policies) {
          return;
        }
        const {requireNumbers, requireLowercase, requireUppercase, requireSymbols} = (cfnUserPool.policies as CfnUserPool.PoliciesProperty)
            .passwordPolicy as CfnUserPool.PasswordPolicyProperty;
          return JSON.stringify([
            ...(requireNumbers ? ['REQUIRES_NUMBERS']: []),
            ...(requireLowercase ? ['REQUIRES_LOWERCASE'] : []),
            ...(requireUppercase ? ['REQUIRES_UPPERCASE'] : []),
            ...(requireSymbols ? ['REQUIRES_SYMBOLS'] : []),
        ]);
```

Reply:
I updated the return statement, but the 2nd portion of this suggestion seems a bit hacky so I just left the original for that